zerver/actions/realm_settings.py

```diff
@@ -186,13 +186,26 @@ def do_set_realm_message_editing(
     send_event(realm, event, active_user_ids(realm.id))
 
 
-def do_set_realm_notifications_stream(
-    realm: Realm, stream: Optional[Stream], stream_id: int, *, acting_user: Optional[UserProfile]
+def do_set_realm_stream(
+    realm: Realm,
+    field: str,
+    stream: Optional[Stream],
+    stream_id: int,
+    *,
+    acting_user: Optional[UserProfile],
 ) -> None:
-    old_value = realm.notifications_stream_id
-    realm.notifications_stream = stream
+    # note: these strings are carefully chosen (and not synthesized)
+    # to enable searching the codebase for these field names.
```

Comment:
I'm not really sure what you mean by "synthesized" here.  All of the occurrences of the string `"notifications_stream"` are indeed relating to the field with that name, but that's not all that surprising.

Reply:
I rewrote this comment before merging.